File: src/game/events/ExplosionLengthPowerUpEvent.java
package game.events;

import game.Bomberman;


/**
 * Fires a game event using Strategy Pattern;
 */
public class ExplosionLengthPowerUpEvent implements GameEvent {
    @Override
    public void invoke(Object arg) {
        Bomberman.getMatch().getCurrentLevel().explosionLengthPowerUpEvent();
    }
}

File: src/game/engine/events/game/ExplosionLengthPowerUpEvent.kt
package game.engine.events.game

import game.Bomberman
import game.engine.world.domain.entity.actors.impl.bomber_entity.base.BomberEntity
import game.engine.events.models.GameEvent


class ExplosionLengthPowerUpEvent : GameEvent {
    override fun invoke(arg: Any?) {
        Bomberman.getMatch().currentLevel!!.eventHandler.onUpdateBombsLengthEvent(arg as BomberEntity, arg.currExplosionLength + 1)
    }
}